### ISSUE TITLE
BlocksCanBuildEvent issue on Paper 1.12.2

### ISSUE THREAD

**Issue by explosiv4** [kernitus/BukkitOldCombatMechanics#731]
## Information

* Server Version: Paper 1.12.2

* OldCombatMechanics version: 2.0.2-beta

* Server Log File: 
<details><summary>Server Log</summary>
  
``` console
[00:15:18] [Server thread/ERROR]: Could not pass event BlockCanBuildEvent to OldCombatMechanics v2.0.1
java.lang.NoSuchMethodError: 'org.bukkit.entity.Player org.bukkit.event.block.BlockCanBuildEvent.getPlayer()'
	at kernitus.plugin.OldCombatMechanics.module.ModuleSwordBlocking.onBlockPlace(ModuleSwordBlocking.java:47) ~[?:?]
	at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor95.execute(Unknown Source) ~[?:?]
	at org.bukkit.plugin.EventExecutor$2.execute(EventExecutor.java:70) ~[patched_1.12.2.jar:git-Paper-1620]
	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:78) ~[patched_1.12.2.jar:git-Paper-1620]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[patched_1.12.2.jar:git-Paper-1620]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:513) ~[patched_1.12.2.jar:git-Paper-1620]
	at net.minecraft.server.v1_12_R1.World.a(World.java:2781) ~[patched_1.12.2.jar:git-Paper-1620]
	at net.minecraft.server.v1_12_R1.ItemBlock.a(ItemBlock.java:23) ~[patched_1.12.2.jar:git-Paper-1620]
	at net.minecraft.server.v1_12_R1.ItemStack.placeItem(ItemStack.java:191) ~[patched_1.12.2.jar:git-Paper-1620]
	at net.minecraft.server.v1_12_R1.PlayerInteractManager.a(PlayerInteractManager.java:551) ~[patched_1.12.2.jar:git-Paper-1620]
	at net.minecraft.server.v1_12_R1.PlayerConnection.a(PlayerConnection.java:1011) ~[patched_1.12.2.jar:git-Paper-1620]
	at net.minecraft.server.v1_12_R1.PacketPlayInUseItem.a(PacketPlayInUseItem.java:37) ~[patched_1.12.2.jar:git-Paper-1620]
	at net.minecraft.server.v1_12_R1.PacketPlayInUseItem.a(PacketPlayInUseItem.java:5) ~[patched_1.12.2.jar:git-Paper-1620]
	at net.minecraft.server.v1_12_R1.PlayerConnectionUtils.lambda$ensureMainThread$0(PlayerConnectionUtils.java:14) ~[patched_1.12.2.jar:git-Paper-1620]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572) ~[?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:317) ~[?:?]
	at net.minecraft.server.v1_12_R1.SystemUtils.a(SourceFile:46) ~[patched_1.12.2.jar:git-Paper-1620]
	at net.minecraft.server.v1_12_R1.MinecraftServer.D(MinecraftServer.java:850) ~[patched_1.12.2.jar:git-Paper-1620]
	at net.minecraft.server.v1_12_R1.DedicatedServer.D(DedicatedServer.java:423) ~[patched_1.12.2.jar:git-Paper-1620]
	at net.minecraft.server.v1_12_R1.MinecraftServer.C(MinecraftServer.java:774) ~[patched_1.12.2.jar:git-Paper-1620]
	at net.minecraft.server.v1_12_R1.MinecraftServer.run(MinecraftServer.java:666) ~[patched_1.12.2.jar:git-Paper-1620]
	at java.lang.Thread.run(Thread.java:1583) [?:?]
```
</details>

* OldCombatMechanics config file:  default one
<details><summary>config.yml</summary>
  
``` yaml

```
</details>

## Problem Description

When i place block this error message is sent to console.

### To Reproduce
 
1.  Place block
2. Check console

### Expected Behaviour

Nothing in console

### Actual Behaviour

Console error message

**Comment by SavageGamer69** [comment]
i have a very similar error on my paper 1.12.2 server, using OCM version 2.0.1:

config.yml:
<details>
<summary>config.yml</summary>

https://drive.google.com/file/d/1lwuDWBtyaRy-nOiAnefXvMimrR8O8gv4/view?usp=sharing (can't directly paste it here because formatting, sorry)

</details>
error:
<details>
<summary>OCM error:</summary>

```
- [Wed 08:07:17 ERROR Minecraft] Could not pass event BlockCanBuildEvent to OldCombatMechanics v2.0.1
java.lang.NoSuchMethodError: org.bukkit.event.block.BlockCanBuildEvent.getPlayer()Lorg/bukkit/entity/Player;
    at kernitus.plugin.OldCombatMechanics.module.ModuleSwordBlocking.onBlockPlace(ModuleSwordBlocking.java:47)
    at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor413.execute(Unknown Source)
    at org.bukkit.plugin.EventExecutor$2.execute(EventExecutor.java:70)
    at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:78)
    at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62)
    at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:513)
    at net.minecraft.server.v1_12_R1.World.a(World.java:2781)
    at net.minecraft.server.v1_12_R1.ItemBlock.a(ItemBlock.java:23)
    at net.minecraft.server.v1_12_R1.ItemStack.placeItem(ItemStack.java:191)
    at net.minecraft.server.v1_12_R1.PlayerInteractManager.a(PlayerInteractManager.java:551)
    at net.minecraft.server.v1_12_R1.PlayerConnection.a(PlayerConnection.java:1011)
    at net.minecraft.server.v1_12_R1.PacketPlayInUseItem.a(PacketPlayInUseItem.java:37)
    at net.minecraft.server.v1_12_R1.PacketPlayInUseItem.a(PacketPlayInUseItem.java:5)
    at net.minecraft.server.v1_12_R1.PlayerConnectionUtils.lambda$ensureMainThread$0(PlayerConnectionUtils.java:14)
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at net.minecraft.server.v1_12_R1.SystemUtils.a(SourceFile:46)
    at net.minecraft.server.v1_12_R1.MinecraftServer.D(MinecraftServer.java:850)
    at net.minecraft.server.v1_12_R1.DedicatedServer.D(DedicatedServer.java:423)
    at net.minecraft.server.v1_12_R1.MinecraftServer.C(MinecraftServer.java:774)
    at net.minecraft.server.v1_12_R1.MinecraftServer.run(MinecraftServer.java:666)
    at java.lang.Thread.run(Thread.java:750)
```

</details>

Same steps to reproduce.

**Comment by kernitus** [comment]
This was fixed as part of #719. @SavageGamer69 @explosiv4 can you try the [newest test version](https://ci.rayzr.dev/job/OldCombatMechanics/) to see if you still get the error?

**Comment by SavageGamer69** [comment]
> This was fixed as part of #719. @SavageGamer69 @explosiv4 can you try the [newest test version](https://ci.rayzr.dev/job/OldCombatMechanics/) to see if you still get the error?

I will try it as soon as I can.i will update.

**Comment by SavageGamer69** [comment]
@kernitus the new version does fix this issue. thank you.
